packages/core/useResize/index.ts

```diff
@@ -0,0 +1,336 @@
+import { watch, ref, reactive, unref, nextTick, computed } from 'vue-demi'
+import { useEventListener, useResizeObserver } from '@vueuse/core'
+import { createEventHook, Fn, tryOnScopeDispose, MaybeRef } from '@vueuse/shared'
+import { MaybeElementRef, unrefElement } from '../unrefElement'
+import { ConfigurableWindow, defaultWindow } from '../_configurable'
+
+type Edges = 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right' | 'left' | 'right' | 'top' | 'bottom'
+
+export interface UseResizeOptions extends ConfigurableWindow {
+  disabled?: boolean
+  disableResize?: MaybeRef<boolean>
+  disableCursor?: MaybeRef<boolean>
+  xMultiplier?: MaybeRef<number>
+  yMultiplier?: MaybeRef<number>
+  borderRadius?: MaybeRef<number>
+  minWidth?: MaybeRef<number> | 'initial'
+  maxWidth?: MaybeRef<number> | 'initial'
+  minHeight?: MaybeRef<number> | 'initial'
+  maxHeight?: MaybeRef<number> | 'initial'
+  edges?: MaybeRef<Edges[]>
+}
+
+export function useResize(element: MaybeElementRef, options: UseResizeOptions = {}) {
+  const {
+    window = defaultWindow,
+    disabled = false,
+    disableResize = false,
+    disableCursor = false,
+    xMultiplier = 1,
+    yMultiplier = 1,
+    borderRadius = 0,
+    edges = ['top-left', 'top-right', 'bottom-left', 'bottom-right', 'left', 'right', 'top', 'bottom'],
+  } = options
+  let {
+    minWidth = 0,
+    maxWidth = Infinity,
+    minHeight = 0,
+    maxHeight = Infinity,
+  } = options
+
+  const isActive = ref(disabled)
+  const target = computed(() => unrefElement(element))
+
+  let width = 0
+  let height = 0
+
+  const onResizeStart = createEventHook<{ pointer: PointerEvent }>()
+  const onResizeEnd = createEventHook<{ pointer: PointerEvent }>()
+  const onResizeMove = createEventHook<{
+    pointer: PointerEvent
+    xDiff: number
+    yDiff: number
+    startX: number
+    startY: number
+    newWidth: number
+    newHeight: number
+    setSize: typeof setSize
+  }>()
+
+  const isOutside = ref(false)
+  const isOverEdge = ref(false)
+  const isResizing = ref(false)
+  const isPathIncludesTarget = ref(false)
+  const direction = ref('')
+
+  const pointer = reactive({ startX: 0, startY: 0, currentX: 0, currentY: 0 })
+
+  const widthRef = ref(0)
+  const heightRef = ref(0)
+
+  let cleanup: Fn[] = []
+
+  const start = () => {
```

Comment:
I think we can get rid of the start function and cleanup array. We can move the useEventListener to the top level as well as everything else. This was actually the source of a bug I ran into. When I had two resizable elements on the page and removed one, then the other one would lose its event handlers as well for some reason. Not sure the source of this, but moving them to the top level fixed this.

Reply:
We will also want to wrap the events in 
```
if (isClient) {
  ...
}
```

Reply:
This is definitely a bug. I will look into it.